src/nfc/clf/pn532.py

```diff
@@ -404,16 +404,40 @@ def init(transport):
                 elif transport.port == "/dev/ttyS0":
                     log.debug("ttyS0 can only do 115.2 kbps")
                     change_baudrate = False  # RPi 'mini uart'
+                    max_baudrate = 115200
 
-        get_version_cmd = bytearray.fromhex("0000ff02fed4022a00")
-        get_version_rsp = bytearray.fromhex("0000ff06fad50332")
-        transport.write(long_preamble + get_version_cmd)
-        log.debug("wait %d ms for data on %s", initial_timeout, transport.port)
-        if not transport.read(timeout=initial_timeout) == Chipset.ACK:
-            raise IOError(errno.ENODEV, os.strerror(errno.ENODEV))
-        if not transport.read(timeout=100).startswith(get_version_rsp):
+        found = False
+        current_baudrate = 0
+        for b in Chipset.br:
+            if b > max_baudrate or b < default_baudrate:
+                continue
+
+            log.debug("Trying baudrate %d", b)
+
+            transport.baudrate = b
+
+            try:
+                get_version_cmd = bytearray.fromhex("0000ff02fed4022a00")
+                get_version_rsp = bytearray.fromhex("0000ff06fad50332")
+                transport.write(long_preamble + get_version_cmd)
+                log.debug("wait %d ms for data", initial_timeout)
+                if not transport.read(timeout=initial_timeout) == Chipset.ACK:
+                    log.debug("read ack error")
+                    continue
+                if not transport.read(timeout=100).startswith(get_version_rsp):
+                    log.debug("get_version_rsp error")
+                    continue
+            except:
+                continue
```

Comment:
This should probably at least log a message, rather than silently eating up *any* exception. Is there any particular exception you are expecting here?